packages/lab/src/tabs-next/TabNext.tsx

```diff
@@ -0,0 +1,159 @@
+// TODO close button needs to be a button. Hence tab needs to include 2 buttons
+import { Button, ButtonProps, makePrefixer, useForkRef } from "@salt-ds/core";
+import { CloseIcon, CloseSmallIcon } from "@salt-ds/icons";
+import { clsx } from "clsx";
+import {
+  forwardRef,
+  ForwardedRef,
+  KeyboardEvent,
+  MouseEvent,
+  ReactElement,
+  useCallback,
+  useRef,
+  useState,
+} from "react";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { TabProps } from "./TabsNextTypes";
+
+import tabCss from "./TabNext.css";
+
+const withBaseName = makePrefixer("saltTabNext");
+
+//TODO not ideal - duplicating the Icon then hiding one in css based on density - is there a nicer way ?
+const CloseTabButton = (props: ButtonProps) => (
+  // FIXME: use polymorphic button
+  <Button
+    {...props}
+    aria-label="Close Tab (Delete or Backspace)"
+    className={withBaseName("closeButton")}
+    tabIndex={-1}
+    title="Close Tab (Delete or Backspace)"
+    variant="secondary"
+  >
+    <CloseIcon
+      aria-label="Close Tab (Delete or Backspace)"
+      className={withBaseName("close-icon")}
+    />
+    <CloseSmallIcon
+      aria-label="Close Tab (Delete or Backspace)"
+      className={withBaseName("close-icon-small")}
+    />
+  </Button>
+);
+
+export const TabNext = forwardRef(function Tab(
+  {
+    ariaControls,
+    children,
+    className,
+    closeable,
+    dragging,
+    index,
+    label,
+    onClick,
+    onClose,
+    onFocus: onFocusProp,
+    onKeyDown,
+    onKeyUp,
+    onMouseDown,
+    selected,
+    tabChildIndex = 0,
+    tabIndex,
+    ...props
+  }: TabProps,
+  ref: ForwardedRef<HTMLDivElement>
+): ReactElement<TabProps> {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-tab-next",
+    css: tabCss,
+    window: targetWindow,
+  });
+  if (index === undefined || onClick === undefined || onKeyDown === undefined) {
```

Comment:
```suggestion
  if (index === undefined || onClick === undefined || onKeyDown === undefined || onKeyUp === undefined ) {
```

error msg seems to indicate `onKeyUp` is required